src/command_line_parser.cc

```diff
@@ -1398,16 +1416,11 @@ TritonParser::Parse(int argc, char** argv)
               ParseOption<int>(optarg);
           break;
         case OPTION_GRPC_RESTRICTED_PROTOCOL: {
-          const auto& parsed_tuple = ParseGrpcRestrictedProtocolOption(optarg);
-          const auto& protocols = SplitOptions(std::get<0>(parsed_tuple), ",");
-          const auto& key = std::get<1>(parsed_tuple);
-          const auto& value = std::get<2>(parsed_tuple);
-          grpc::ProtocolGroup pg;
-          for (const auto& p : protocols) {
-            pg.protocols_.emplace(p);
-          }
-          pg.restricted_key_ = std::make_pair(key, value);
-          lgrpc_options.protocol_groups_.emplace_back(pg);
+          ParseRestrictedFeatureOption(
+              optarg, long_options[option_index].name,
+              std::string(
+                  triton::server::grpc::kRestrictedProtocolHeaderTemplate),
+              "protocol", lgrpc_options.restricted_protocols_);
```

Comment:
1. Why is feature_name "protocol" for grpc and "api" for http? 
2. Can http_options and grpc_options have the same member variable name, either `.restricted_protocols_` or `.restricted_apis_` rather than being different?

Reply:
Mainly just to align with the command line setting / documentation. gRPC specifically calls out protocols throughout and that didn't make sense for HTTP - where the closest seemed to be API. 

could make them both restricted_features_ if clearer -

Reply:
Oh sorry, I wouldn't to change the `feature_type` from "protocol" as its already part of the existing docs / help messages. So for printing / documentation I think keeping it "protocol" and "api" makes sense. 

For the internal variable I think it matters less.

Reply:
both restricted_features_ LGTM

Reply:
deferring name change to documentation PR / larger discussion on if it's ok to change from grpc protocol